iox_client_meta_schema.go

```diff
@@ -0,0 +1,110 @@
+package influxdbiox
+
+import (
+	"context"
+	"errors"
+
+	schema "github.com/influxdata/influxdb-iox-client-go/internal/schema"
+)
+
+// ColumnType defines the column data types IOx can represent.
+type ColumnType int32
+
+const (
+	// ColumnTypeUnknown is an invalid column type.
+	ColumnTypeUnknown ColumnType = 0
+	// ColumnType_I64 is an int64.
+	ColumnType_I64 ColumnType = 1
+	// ColumnType_U64 is an uint64.
+	ColumnType_U64 ColumnType = 2
+	// ColumnType_F64 is an float64.
+	ColumnType_F64 ColumnType = 3
+	// ColumnType_BOOL is a bool.
+	ColumnType_BOOL ColumnType = 4
+	// ColumnType_STRING is a string.
+	ColumnType_STRING ColumnType = 5
+	// ColumnType_TIME is a timestamp.
+	ColumnType_TIME ColumnType = 6
+	// ColumnType_TAG is a tag value.
+	ColumnType_TAG ColumnType = 7
+)
+
+func (c ColumnType) String() string {
+	switch c {
+	case ColumnType_I64:
+		return "int64"
+	case ColumnType_U64:
+		return "uint64"
+	case ColumnType_F64:
+		return "float64"
+	case ColumnType_BOOL:
+		return "bool"
+	case ColumnType_STRING:
+		return "string"
+	case ColumnType_TIME:
+		return "timestamp"
+	case ColumnType_TAG:
+		return "tag"
+	default:
+		return "unknown"
+	}
+}
+
+// Return a map of column name to data types for the specified table in
+// namespace.
+func (c *Client) GetSchema(ctx context.Context, namespace string, table string) (map[string]ColumnType, error) {
```

Comment:
Just to be clear, `namespace` here is the database name, which is `<org_id>_<bucket_id>`, right?

Reply:
Yes sorry - an extra mental model conversion when switching between IOx/everything else - namespace is indeed a `org_bucket`, though it is not constrained to that format.